src/absorbable.rs

```diff
@@ -0,0 +1,330 @@
+use ark_ec::models::short_weierstrass_jacobian::GroupAffine as SWAffine;
+use ark_ec::models::twisted_edwards_extended::GroupAffine as TEAffine;
+use ark_ec::models::{SWModelParameters, TEModelParameters};
+use ark_ff::models::{
+    Fp256, Fp256Parameters, Fp320, Fp320Parameters, Fp384, Fp384Parameters, Fp768, Fp768Parameters,
+    Fp832, Fp832Parameters,
+};
+use ark_ff::{to_bytes, PrimeField, ToConstraintField};
+use ark_std::vec;
+use ark_std::vec::Vec;
+
+/// An interface for objects that can be absorbed by a `CryptographicSponge`.
+pub trait Absorb<F: PrimeField> {
+    /// Converts the object into a list of bytes that can be absorbed by a `CryptographicSponge`.
+    fn to_sponge_bytes(&self) -> Vec<u8>;
+
```

Comment:
I don't find any use case of `to_sponge_bytes` in sponge. Poseidon just use `to_sponge_field_elements`. Will it be useful in the future?

Reply:
It's probably useful for byte-based sponges, like SHA2 or Blake2s

Reply:
Makes sense